geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/data/RedisSortedSet.java

```diff
@@ -228,6 +230,26 @@ private void validateScoreIsDouble(byte[] score) {
     return members.get(member);
   }
 
+  byte[] zincrby(Region<RedisKey, RedisData> region, RedisKey key, byte[] increment,
+      byte[] member) {
+    byte[] score = members.get(member);
+    byte[] incr = Coder.doubleToBytes(
+        processIncrement(Coder.bytesToString(increment).toLowerCase()));
+
+    if (score != null) {
+      incr = Coder.doubleToBytes(Coder.bytesToDouble(score) + Coder.bytesToDouble(incr));
+    }
```

Comment:
I imagine you could check the `incr` variable after for `NaN`, something like this, though there might be a better way of doing it:
```
 if(Arrays.equals(incr, "NaN".getBytes())) {
      throw new NumberFormatException(ERROR_OPERATION_PRODUCED_NAN);
  }
 ```

Reply:
making that test change you suggested showed that not doing that check is a bug. I implemented the check.